extension/test/e2e/klarna.spec.js

```diff
@@ -153,42 +164,52 @@ describe('::klarnaPayment::', () => {
     // Submit payment details
     const returnPageUrl = new URL(browserTab.url())
     const searchParamsJson = Object.fromEntries(returnPageUrl.searchParams)
-
-    const { body: updatedPayment } = await ctpClient.update(
-      ctpClient.builder.payments,
-      payment.id,
-      payment.version,
-      [
-        {
-          action: 'setCustomField',
-          name: 'submitAdditionalPaymentDetailsRequest',
-          value: JSON.stringify({
-            details: searchParamsJson,
-          }),
-        },
-      ]
-    )
-
-    return updatedPayment
+    let result = null
+    try {
+      result = await ctpClient.update(
+        ctpClient.builder.payments,
+        payment.id,
+        payment.version,
+        [
+          {
+            action: 'setCustomField',
+            name: 'submitAdditionalPaymentDetailsRequest',
+            value: JSON.stringify({
+              details: searchParamsJson,
+            }),
+          },
+        ]
+      )
+    } catch (err) {
+      logger.error('klarna::handleRedirect::errors:', err.body.errors)
+    }
+    return result.body
   }
 
   async function capturePayment({ payment }) {
     const transaction = payment.transactions[0]
-    const { body: updatedPayment } = await ctpClient.update(
-      ctpClient.builder.payments,
-      payment.id,
-      payment.version,
-      [
-        pU.createAddTransactionAction({
-          type: 'Charge',
-          state: 'Initial',
-          currency: transaction.amount.currencyCode,
-          amount: transaction.amount.centAmount,
-        }),
-      ]
-    )
-
-    return updatedPayment
+    let result = null
+    try {
+      result = await ctpClient.update(
+        ctpClient.builder.payments,
+        payment.id,
+        payment.version,
+        [
+          pU.createAddTransactionAction({
+            type: 'Charge',
+            state: 'Initial',
+            currency: transaction.amount.currencyCode,
+            amount: transaction.amount.centAmount,
+          }),
+        ]
+      )
+    } catch (err) {
+      logger.error(
+        'klarna::capturePayment::errors:',
+        JSON.stringify(err.body.errors)
```

Comment:
Why do you have sometimes `JSON.stringify(err.body.errors)` and sometimes only `err.body.errors`?

Reply:
It's suggested to stringify the object. Otherwise` \n` is added into the JSON, which make the printed out JSON unformattable and hard to read.
https://github.com/commercetools/commercetools-adyen-integration/pull/735/commits/f11be03a96ad9da558e733382117598cc9b93c7b